packages/client/tests/functional/_utils/setupTestSuiteClient.ts

```diff
@@ -317,5 +317,24 @@ export function setupTestSuiteClientDriverAdapter({
     }
   }
 
+  if (driverAdapter === 'js_mariadb') {
+    const { PrismaMariaDb } = require('@prisma/adapter-mariadb') as typeof import('@prisma/adapter-mariadb')
+
+    const [, user, password, host, port, database] =
+      datasourceInfo.databaseUrl.match(/^mysql:\/\/([^:]+):([^@]+)@([^:;]+):(\d+)\/(.+)$/) || []
```

Comment:
Doesn't matter much in tests since we control the connection string and know the regex will work for it, but might've been easier to use `URL` to parse it:

```suggestion
    const url = new URL(datasourceInfo.databaseUrl)
    const { username: user, password, hostname: host, port } = url
    const database = url.pathname && url.pathname.slice(1)
```